### PR TITLE
implementation of dark mode in individual packages

### LINKED ISSUE #699: Implement dark mode for our storybooks
## Description
We could have a dark mode for our 3 storybooks. There are many plug-ins available for the same.

- https://storybook.js.org/docs/react/configure/theming
- https://github.com/hipstersmoothie/storybook-dark-mode

(The above links are for reference sake only)

What is needed, perhaps, is a button which toggles state from light to dark thereby changing the theme of the entire storybook. 

## Implementation

- [ ] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by neeraj-2**
## Fixes

Fixes #699 by @nimishbongale 

## Description
Implements toggling b/w Dark mode and Light mode in the 3 storybooks `Fonts`, `Vocabulary` and `Vue-Vocabulary` 

## Demo:

ezgif com-gif-maker

sun icon
moonicon

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: packages/fonts/.storybook/manager.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
 
-import theme from './theme'
+import { themes } from '@storybook/theming'
+import 'storybook-dark-mode/register'
 
 addons.setConfig({
-  theme
-})
+  theme:themes.light
+})
```
Comment (nimishbongale) [inline packages/fonts/.storybook/manager.js]: Can we add a newline here? Looks cleaner imo. Formatting wise.

File: packages/vocabulary/.storybook/manager.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 
-import theme from './theme'
+import { themes } from '@storybook/theming'
+import 'storybook-dark-mode/register'
 
 addons.setConfig({
-  theme
-})
+  theme:themes.light
+})
```
Comment (nimishbongale) [inline packages/vocabulary/.storybook/manager.js]: Here as well.

File: packages/fonts/package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "sass-loader": "^10.0.2",
     "sass-true": "^5.0.0",
     "start-server-and-test": "^1.10.11",
+    "storybook-dark-mode": "^1.0.3",
```
Comment (nimishbongale) [inline packages/fonts/package.json:44]: 👍

File: packages/vocabulary/package.json [hunk 4]
```diff
@@ -47,6 +47,7 @@
     "sass-true": "^5.0.0",
     "start-server-and-test": "^1.10.11",
     "storybook-addon-designs": "^5.4.1",
+    "storybook-dark-mode": "^1.0.3",
```
Comment (nimishbongale) [inline packages/vocabulary/package.json:50]: 👍

File: packages/vue-vocabulary/.storybook/manager.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
...
 
-import theme from './theme'
+import { themes } from '@storybook/theming'
+import 'storybook-dark-mode/register'
 
 addons.setConfig({
-  theme
-})
+  theme:themes.light
+})
```
Comment (nimishbongale) [inline packages/vue-vocabulary/.storybook/manager.js]: Here as well.

File: packages/vue-vocabulary/package.json [hunk 6]
```diff
@@ -57,6 +57,7 @@
     "react": "^16.13.1",
     "react-is": "^16.13.1",
     "storybook-addon-designs": "^5.4.1",
+    "storybook-dark-mode": "^1.0.3",
```
Comment (nimishbongale) [inline packages/vue-vocabulary/package.json:60]: 👍

**Review by nimishbongale — CHANGES_REQUESTED** [review]
Great work @neeraj-2 ! 🚀 

Couple of things:

1. Maybe it would be good idea if the storybooks started out on a light theme, and then upon button click toggling to the dark mode. Let the light mode be the default one.

2. Is it possible to reposition that toggle button to the end of the tab bar?

Also @neeraj-2 you can go ahead and raise a normal PR after this. The build passes, and the changes I've suggested can be incorporated by you in the normal PR.

File: packages/fonts/.storybook/manager.js (hunk 1 above)
Comment (neeraj-2) [reply packages/fonts/.storybook/manager.js]: After the 7th line, right?

Comment (nimishbongale) [reply packages/fonts/.storybook/manager.js]: After 8th.

There's a convention to have an endline at the end of file.

Comment (neeraj-2) [reply packages/fonts/.storybook/manager.js]: Sure 👍

Comment (nimishbongale) [reply packages/fonts/.storybook/manager.js]: You'd need to do that for some other files as well. I've commented wherever the change needs to be done.

**Comment by neeraj-2** [comment]
> Great work @neeraj-2 ! 🚀
> 
> Couple of things:
> 
> 1. Maybe it would be good idea if the storybooks started out on a light theme, and then upon button click toggling to the dark mode. Let the light mode be the default one.
> 2. Is it possible to reposition that toggle button to the end of the tab bar?
> 
> Also @neeraj-2 you can go ahead and raise a normal PR after this. The build passes, and the changes I've suggested can be incorporated by you in the normal PR.

I will try to make the light theme as default, but I don't know if it is possible to reposition that toggle button to the end of the tab bar?

**Comment by nimishbongale** [comment]
> I will try to make the light theme as default, but I don't know if it is possible to reposition that toggle button to the end of the tab bar?

- https://raw.githubusercontent.com/hipstersmoothie/storybook-dark-mode/master/example.gif

If you look closely here, they have it at the end of the tab bar.

**Comment by neeraj-2** [comment]
> > Great work @neeraj-2 ! 🚀
> > Couple of things:
> > 
> > 1. Maybe it would be good idea if the storybooks started out on a light theme, and then upon button click toggling to the dark mode. Let the light mode be the default one.
> > 2. Is it possible to reposition that toggle button to the end of the tab bar?
> > 
> > Also @neeraj-2 you can go ahead and raise a normal PR after this. The build passes, and the changes I've suggested can be incorporated by you in the normal PR.
> 
> I will try to make the light theme as default, but I don't know if it is possible to reposition that toggle button to the end of the tab bar?

 @nimishbongale I found that the initial theme is set to light , but if some user set dark theme and reload it, then the theme would be the dark one. Implies, whatever the theme user has selected, after reload we will get that theme only. So do we want that even after any reload whatever the theme earlier, still the theme should be light one?

**Comment by nimishbongale** [comment]
@neeraj-2 I'm saying I'd want the defaukt

**Comment by nimishbongale** [comment]
@neeraj-2 I'm saying that if a user hasn't selected dark mode initially, it should open in light mode. Then the users choice will be preferred.

**Comment by neeraj-2** [comment]
> @neeraj-2 I'm saying that if a user hasn't selected dark mode initially, it should open in light mode. Then the users choice will be preferred.
Great then, it is the behavior in the previous commits. Now the only requirement is to reposition the tab bar.So, I am focussing on that

**Comment by nimishbongale** [comment]
> Great then, it is the behavior in the previous commits. Now the only requirement is to reposition the tab bar.So, I am focussing on that

@neeraj-2 

I'm not sure if the default behaviour is that way.

Head over [here](https://deploy-preview-723--cc-vocabulary.netlify.app/) (perhaps in incognito) to see your live code running. It by default opens the storybook in dark mode. I'd suggest it to open in light mode, then transition depending on user input.

**Comment by neeraj-2** [comment]
> > Great then, it is the behavior in the previous commits. Now the only requirement is to reposition the tab bar.So, I am focussing on that
> 
> @neeraj-2
> 
> I'm not sure if the default behaviour is that way.
> 
> Head over [here](https://deploy-preview-723--cc-vocabulary.netlify.app/) (perhaps in incognito) to see your live code running. It by default opens the storybook in dark mode. I'd suggest it to open in light mode, then transition depending on user input.

@nimishbongale 
I opened the link in incognito, I found, it opened the storybook in light mode. Is this not the case with you. Is this something related to the browser? Btw I am using `Google chrome`

**Comment by nimishbongale** [comment]
Okay I'm not entirely sure why that happens, but I'm guessing it takes the default theme that is set in my chrome browser. I have dark mode enabled everywhere.

Okay @neeraj-2 you can go ahead and raise a normal PR with the changes as suggested. Let's get this merged 🚀

**Comment by neeraj-2** [comment]
> Okay I'm not entirely sure why that happens, but I'm guessing it takes the default theme that is set in my chrome browser. I have dark mode enabled everywhere.
> 
> Okay @neeraj-2 you can go ahead and raise a normal PR with the changes as suggested. Let's get this merged 🚀

Sure @nimishbongale 👍

**Pull request closed without merging**